lib/chai/core/assertions.js

```diff
@@ -988,6 +988,13 @@ module.exports = function (chai, _) {
 
   function assertInstanceOf (constructor, msg) {
     if (msg) flag(this, 'message', msg);
+    var isFunction = constructor instanceof Function;
+
+    if (!isFunction) {
```

Comment:
I think that it would be an overkill. In this case we just need to know whether something is a function or not in order to see if it is a constructor, there's no need for anything more complex than that.

Please let me know if I forgot any edge case, but I think we should go native if we can.